src/AgDatabase.cs

```diff
@@ -79,6 +80,8 @@ public AgDatabase(DatabaseConfig dbConfig)
 
     public int ServerRemainingDiskMb => _listener.Primary.RemainingDiskMb();
 
+    public IListener Listener => _listener;
```

Comment:
Instead of this accessor to a private member variable, it's a bit more clear to make this a property
https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/using-properties#the-get-accessor

The [auto implemented property](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/auto-implemented-properties) example that you'd use here:
`public IListener Listener { get; }`